src/components/deploymentConfig/DeploymentTemplateView/DeploymentTemplateGUIView.tsx

```diff
@@ -124,6 +126,9 @@ const DeploymentTemplateGUIView = ({
                 onChange={handleFormChange}
                 uiSchema={UISchema}
                 disabled={readOnly}
+                experimental_defaultFormStateBehavior={{
```

Comment:
This will prevent the empty states from being added whenever we interact with the form.

For example suppose autoscaling does not exist in advanced yaml but exists in guiSchema. Suppose we make change to some other field that is in both guiSchema and advanced yaml - let's take it to be ContainerPort. Then when we switch to advanced we will see an object of autoscaling with empty values for its property fields